### PR TITLE
Adds Earn Your Beard (ebeards)

### PR DESCRIPTION
**Pull request by JJRcop**
:cl:
add: Adds Earn Your Beard. Enabled in character preferences, this forces you to grow your beard out by surviving consecutive rounds in the same body, in exchange for cool animated effects!
/:cl:

If enabled in preferences, you will start shaven, and will grow your beard as you survive consecutive rounds in a row. The benefit though, is your beard will have an animated effect which is choosable from preferences, that is not attainable anywhere else.

Survival is defined by escaping on the shuttle alive in the exact same body you were spawned in. If the round ends without the shuttle being called, like if you kill the blob, survival is defined as above except without the shuttle part.

rainbow shadow electric blaze

### CODE REVIEW

**Comment by Jalleo** [comment]
Need a image for each animated effect.

**Comment by Shadowlight213** [comment]
The people with the biggest beards will be those who hid in an escape pod roundstart probably.

**Comment by jughu** [comment]
1+ for rainbow beards

**Comment by Jalleo** [comment]
And those that want to kill them will be looking there.

**Comment by jughu** [comment]
Those that just want to rip their beard off and torture them

**Comment by Zilenan91** [comment]
Can you make this forced for everyone so the people with massive beards are respected like old wise men and then killed for their facial hair

**Comment by jughu** [comment]
Give the clown a razor blade to remove beards and then gain their beards with the magic of glue honkhonk

**Comment by WJohn** [comment]
What happens if you die and are cloned in the same round? Do you keep the beard?

**Comment by Thunder12345** [comment]
No your new body comes out clean shaven and you have to start again

**Comment by jughu** [comment]
Clown should be able to rip their beards off and put it in a glass cage which transfer between rounds to see who got "pranked"

**Comment by Shapsy** [comment]
That flashing rainbow beard looks awful

**Comment by JJRcop** [comment]
The .gif didn't come out very well, I'll see about a better one later.

**Comment by Shapsy** [comment]
I mean more like the idea of walking around with a giant pulsating rainbow colored beard is gross

File: code/modules/client/preferences.dm [hunk 1]
```diff
@@ -55,6 +62,8 @@ var/list/preferences_datums = list()
 	var/hair_color = "000"				//Hair color
 	var/facial_hair_style = "Shaved"	//Face hair type
 	var/facial_hair_color = "000"		//Facial hair color
+	var/beard_level_enabled = FALSE		//Enables beard level
```
Comment (Iamgoofball) [inline code/modules/client/preferences.dm]: default to enabled

no more opt in shit

**Comment by MMMiracles** [comment]
Beards will save /tg/.

**Comment by Supermichael777** [comment]
Pets were bad enough

**Comment by KingDairy** [comment]
glorious knee-length beard for wizards and the most robust players when

File: code/modules/client/preferences.dm (hunk 1 above)
Comment (Zilenan91) [inline code/modules/client/preferences.dm]: I dream of a world where beards are respected, where only the most robust or cowardly can have a knee-lengther

**Comment by Chocobro** [comment]
is it specifically consecutive rounds? or just rounds that player plays?

bit of a better explanation: if i survive a round and then don't play for a week, will my beard grow in the next round i play?

**Comment by JJRcop** [comment]
Rounds that the character plays actually, it's per-character slot. But yeah, it only takes effect when you spawn in the round using latejoin or ready up.

(Also, spawning as a cyborg or AI will cause Earn Your Beard to not apply for that round, because neither of those have beards)

**Comment by Zilenan91** [comment]
Does this effect lizards or only humans

**Comment by Chocobro** [comment]
lizards should not have hair

**Comment by JJRcop** [comment]
i have not tested on lizards it might actually work i don't know

**Comment by Zilenan91** [comment]
I for one am for lizard beards if they work

**Comment by JJRcop** [comment]
i do know that it working on females is an intended feature, the only way females could have beards is by earning them

**Comment by Chocobro** [comment]
of course you're for it

**Comment by Chocobro** [comment]
in case the implications behind that weren't clear enough
i'm calling you a lizard fucker

**Comment by JJRcop** [comment]
confirmed it does not work for lizards

**Comment by KorPhaeron** [comment]
I don't think this is going to mesh well with how our admins like to act (lots of buttons that often kill large numbers of people). It's just gonna make people super mad when it happens.

**Comment by Zilenan91** [comment]
That sounds more like an admin problem rather than the code being at fault.

**Comment by KorPhaeron** [comment]
Also I think this might encourage more early shuttle calls/more powergaming because people will want to avoid risks and losing their streak.

**Comment by Zilenan91** [comment]
It's rainbow beards, people shouldn't take it that seriously

**Comment by KorPhaeron** [comment]
It's not just the beard, it's the score/streak behind it.

>caring about scores is dumb!!

Doesn't matter, people love scores. This is like greentext that can last indefinitely.

**Comment by Zilenan91** [comment]
I can't speak for anyone but myself but I would turn this on and if I somehow through some miracle got to a rainbow beard I'd consider it as something momentarily cool then forget about it. Like a pleasant surprise.

**Comment by JJRcop** [comment]
think this should just stay testmerged for a while and we'll see how it goes

**Comment by PKPenguin321** [comment]
Getting greentext should always grow your beard even if you die so you can still do the Die a Glorious Death objective.

**Comment by GunHog** [comment]
I am no longer against this feature. It looks silly, of course, but we have sooooooooooooooo much worse already!

**Comment by Bawhoppen** [comment]
I still hate the fact a person can't have a beard on their character if they want one.

**Comment by GunHog** [comment]
The test merge allowed me to keep Magni's beard so long as I had the "Earn your beard" option off. Has that behavior been changed?

**Comment by JJRcop** [comment]
If you have the 'Earn Your Beard' feature turned off everything will function as before. This is entirely an opt in feature.

**Comment by Bawhoppen** [comment]
I DELETED MY COMMENT GOD SORRY

**Comment by Zilenan91** [comment]
I look forward to shaming people who don't join the beard master race.

**Comment by Robustin** [comment]
Make the rainbow beard less of an eye cancer and reduce the requirements for most beards. Needing 3 rounds just to get a 5 o'clock shadow is pretty absurd.

**Comment by MMMiracles** [comment]
^baby-face who wants ez-beards.

Plasmamen were merely the appetizer, beard survival is the real hardcore mode of SS13.

**Comment by Man-Shroom-** [comment]
shit he got me

**Comment by lordpidey** [comment]
What about if you spawn as a nukeop?  If you don't make it on the shuttle as a nukeop, will you lose your beard?  Because that's kinda dumb.

**Comment by Zilenan91** [comment]
You die if you don't make it on the shuttle so you know. You just don't succeed.

**Comment by Man-Shroom-** [comment]
if you are a nukeop and survive till round end, regardless of where you are, you keep your beard

File: code/modules/client/preferences.dm (hunk 1 above)
Comment (JJRcop) [inline code/modules/client/preferences.dm]: I'm gonna say no and then maybe make a future pr that forces you to grow your beard without any cool effects just for you goofball

**Comment by JJRcop** [comment]
actually according to the code, you have to be back at syndie base to keep your beard

**Comment by GunHog** [comment]
What if you made it only count **non-antag** deaths, so we do not have coward and "friendly" antags increased because of this?

**Comment by Bawhoppen** [comment]
sounds good

**Comment by JJRcop** [comment]
ok when i get back from work i'll make it so that if you're an antag your beard level freezes for that round

**Comment by Bawhoppen** [comment]
boo thats boring

**Comment by hornygranny** [comment]
this is nitpicky but if you survive the round in a cloned body it should set you to beard level 1, not 0

**Comment by KorPhaeron** [comment]
ADMIN LOG: MimicFaux/(Hadrian Melik) modified [0x21017721] (/datum/preferences)'s beard_level to 5

Should add this to the protected var list

**Comment by MMMiracles** [comment]
modifying a player's beard level should be on the same level as giving ghost control to megafauna.

no, worse.

**Comment by GunHog** [comment]
Giving megafauna to ghosts should totally happen for the EPIC LULZ

**Comment by optimumtact** [comment]
Just getting sign off on this from designlead/headcoders

**Comment by Zilenan91** [comment]
If admins are allowed to edit players beards then all it'll lead to is them demanding that admins grow their beards for stupid reasons.

**Comment by optimumtact** [comment]
That's a matter of admin policy

**Comment by Iamgoofball** [comment]
design lead also said to lock that var out

**Comment by optimumtact** [comment]
We literally cannot run around designing to prevent admin interaction because there are too many routes around it.

**Comment by GunHog** [comment]
Just log it properly and deadmin the ones that do it unjustly. Let them have the power, and the responsibility to use it appropriately.

**Comment by hornygranny** [comment]
i can think of a ton of ways to cheat beard growth even as a player, it's not feasible to prevent it

**Comment by Chocobro** [comment]
i unsubscribed from this and it's still sending me fucking emails

**Comment by KorPhaeron** [comment]
I think this will encourage behaviors such as

-People playing passively as antag to stay alive. This will be boring in traitor if we have a bunch of people doing nothing, and be incredibly frustrating in conversion modes when your converts decide not to help because they're trying to protect their score. We actually had an adminhelp from a player today asking to please remove his wizard status from him because he didn't want to lose his beard.

-People will play security/heads even less. If someone is willing to give up wizard, the holy grail of the antag lottery, to protect a beard, I can't imagine people like that are going to want to play high value civilian targets.

-Intense salt when a new feature is unbalanced/unfair/whatever and it kills someone.

-Intense salt when an admin hits a dumb button.

Maybe I'm overestimating the impact this will have, but those are my concerns.

**Comment by GunHog** [comment]
All of the legitimate. I personally do not care about having an e-peen score, but the antag thing is why I requested that antag deaths not be counted.

But, this feature causing other murderous features to be nerfed/removed and admins being shackled is simply unacceptable, and after reading Kor's post, I am conflicted now.  There is a post on the forums stating that people needed reasons to desire survival across rounds (https://tgstation13.org/phpBB/viewtopic.php?f=9&t=7265 - it is actually a reference to this feature).

I am not sure it is a good idea any more, because it introduces negative punishment. If you die, you lose your beard, therefore, you should avoid dying. This could hurt head/security players as they are delicious targets, and obviously would hurt antags as well. But, on the other hand, the number of people who care about their beard above all else might be small enough such that it will simply result in a small group of survivalist players griping when they die, small enough that it does not make a significant difference.

TL;DR: This feature could spawn extreme powergamers, or it might just be something nice that lets us laugh at the whining of a powergamer that dies after surviving 10+ rounds in a row.

**Comment by Zilenan91** [comment]
All I know is that if my beard gets longer it'll be cool, and if I'm antag I'll be sure to sweep the remote sections of the station for maximum salt generation

**Comment by Iamgoofball** [comment]
this is a good feature if we make tweaks like antag deaths not counting against your beard

**Comment by jughu** [comment]
How will mindswapping work out though? Would be funny if you steal their beard level

**Comment by optimumtact** [comment]
Yeah we ain't doing this, and if you want someone to blame

https://tgstation13.org/phpBB/viewtopic.php?f=23&t=7334&view=unread#p197428

**Comment by Man-Shroom-** [comment]
i can't wait until this gets into the game it will be so fun

**Comment by optimumtact** [comment]
Local man remains delusional, onion reports.

**Comment by JJRcop** [comment]
It's fine, I'll put up a cleanup PR for testmerge to get rid of the dumb values
